Command/ExportCommand.php

```diff
@@ -0,0 +1,145 @@
+    {
+        $this->logger = $logger;
+        $this->client = $client;
+        $this->dataService = $dataService;
+        $this->instanceId = $instanceId;
+        $this->contentTypeService = $contentTypeService;
+        parent::__construct($logger, $client);
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('ems:contenttype:export')
+            ->setDescription('Export a search result of a content type to a specific format')
+            ->addArgument(
+                'contentTypeName',
+                InputArgument::REQUIRED,
+                'The document\'s content type name to export'
+            )
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'The format of the output: json, xml, merge-json, merge-xml or the id of the content type\' template',
+                'json'
+            )
+            ->addArgument(
+                'query',
+                InputArgument::OPTIONAL,
+                'The query to run',
+                '{}'
+            )
+            ->addArgument(
+                'index',
+                InputArgument::OPTIONAL,
+                'The index to use for the query, it will use the default environemnt index if not defined'
+            )
+            ->addOption(
+                'withBusinessId',
+                null,
+                InputOption::VALUE_NONE,
+                'Replace internal OUUIDs by business values'
+            )
+            ->addArgument(
+                'scrollSize',
+                InputArgument::OPTIONAL,
+                'Size of the elasticsearch scroll request',
+                100
+            )
+            ->addArgument(
+                'scrollTimeout',
+                InputArgument::OPTIONAL,
+                'Time to migrate "scrollSize" items i.e. 30s or 2m',
+                '1m'
+            )
+            ->addOption(
+                'source',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Query used to find elasticsearch records to import',
+                '{"sort":{"_uid":{"order":"asc"}}}'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+
+        $contentTypeName = $input->getArgument('contentTypeName');
+        $scrollSize = $input->getArgument('scrollSize');
+        $scrollTimeout = $input->getArgument('scrollTimeout');
+        $contentType = $this->contentTypeService->getByName($contentTypeName);
+        if (! $contentType instanceof ContentType) {
+            $output->writeln(sprintf("WARNING: Content type named %s not found", $contentType));
+            return null;
+        }
+        $index = $input->getArgument('index');
+        if ($index === null) {
+            $index = $contentType->getEnvironment()->getAlias();
+        }
+
+        $arrayElasticsearchIndex = $this->client->search([
```

Comment:
The business logic of this command should be abstracted away. Use yield in your abstraction if you want to handle the progress bar in the command.

Reply:
Fair point (I'll do it like this in the future)